benches/arrays/primes.ncl

```diff
@@ -0,0 +1,42 @@
+let range
+  | doc "Generate an array of integers in the range [`start`, `end`)."
+  | Num -> Num -> Array Num
+  = fun start end =>
+    if end <= start then 
+      []
+    else
+      array.generate (fun x => x + start) (end - start) 
+in
+
+let is_prime 
+  | doc "Returns true if the argument is a prime number."
+  = fun x => x > 1 && array.all (fun d => x % d != 0) (range 2 (x - 1))
+in
+
+let Prime = fun label value =>
+  if is_prime value then
+    value
+  else
+    contract.blame_with "%{string.from_num value} is not a prime number." label 
```

Comment:
Not very important for a benchmark, but the idiomatic way would be to write 

```nickel
let Prime = contract.from_predicate is_prime in
```

Reply:
I was aware of that. But I thought the error message could not be customized. Would this make for a new library function? i.e. `contract.from_predicate_with`? (the name is not the strongest argument here :P)